### DISCUSSION TITLE
How to invite users to app store connect?

### DISCUSSION THREAD

**Discussion by lucasvanhalst** [fastlane/fastlane#18029] [Q&A]
I've looked at the documentation (https://github.com/fastlane/fastlane/blob/master/spaceship/docs/AppStoreConnect.md), but it doesn't seem to mention how to invite users to app store connect.
It used to be possible like this:
```ruby
    Spaceship::Members.create!(
        firstname: firstname,
        lastname: lastname,
        email_address: email_address,
        roles: roles
    )
```
but that now returns "Not Found"
According to this PR it should be possible, but I don't know how to use it. https://github.com/fastlane/fastlane/pull/17305

**Comment by dinsen** [comment]
Hi @lucasvanhalst 
I think you can use it like this:
```ruby
Spaceship::ConnectAPI::Users.post_user_invitation(
    email: nil,
    first_name: nil,
    last_name: nil,
    roles: [],
    provisioning_allowed: nil,
    all_apps_visible: nil
)
```
But I have not used it before so maybe you can find out by looking at the code here:
https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/connect_api/users/users.rb#L56

**Comment by lucasvanhalst** [comment]
@Dinsen thanks but that didn't seem to work:
`undefined method post_user_invitation' for Spaceship::ConnectAPI::Users:Module`
also tried `Spaceship::ConnectAPI::Users::API.post_user_invitation` but that also gives the same error

**Reply by dinsen** [reply]
I actually think the create method in the user_invitation model is missing.
https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/connect_api/models/user_invitation.rb

When that is added, you will then be able to use `Spaceship::ConnectAPI::UserInvitation.create()`

**Reply by dinsen** [reply]
@lucasvanhalst I just opened a PR https://github.com/fastlane/fastlane/pull/18043 for this.

**Comment by Stiedeman** [comment]
In order to be able to build and submit your iOS app an invitation to your App Store Connect account is required.

Below you can find more details on how to invite users to it.

1. Access the Apple Developer website clicking here. On the top right section, you can click on “Account” to be redirected to the login form.
2. You can then insert your login credentials to sign-in into your Apple Developer Account
3. Once logged in you will see your Apple Developer Account Dashboard. There you can click on “App Store Connect”.
4. Confirm that you would like to access your App Store Connect account by clicking the button in the middle of the screen.
5. Now that you are under your App Store Connect account, click on the “User and Roles” icon.


[mygroundbiz](https://www.mygroundbiz.website/)
